UnitsNet/UnitConverter.cs

```diff
@@ -407,9 +407,9 @@ private static bool TryParseUnit(Type unitType, string unitName, out object unit
 
         private static bool TryGetUnitType(string quantityName, out Type unitType)
         {
-            quantityName += "Unit";
+            var quantityTypeName = quantityName += "Unit"; // ex. LengthUnit
```

Comment:
It should be named `unitTypeName`, quantity would be `Length`.

Edit: I fixed it in the PR.